src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/EnumerableExtensions.cs

```diff
@@ -15,15 +15,10 @@ internal static class EnumerableExtensions
     {
         public static T? FirstOrDefault<T>(this ImmutableArray<T> collection, Func<T, bool> predicate)
         {
-            foreach (var item in collection)
-            {
-                if (predicate(item))
-                {
-                    return item;
-                }
-            }
-
-            return default;
+            // Despite being on the System.Linq namespace, this function resides
+            // in the System.Collections.Immutable assembly and avoids importing
+            // LINQ elsewhere.
+            return ImmutableArrayExtensions.FirstOrDefault(collection, predicate);
```

Comment:
Not getting why do we need this change, from comment in the summary `Replacements for System.Linq to avoid an unnecessary dependency.`  it seems we want to avoid referencing System.Linq.

Reply:
Removed.

Reply:
I am actually meant about the comment above the API:
https://github.com/dotnet/runtime/blob/2685584512dbc6bfeef72abc5adfe9160a9bb4a5/src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/EnumerableExtensions.cs#L10-L13
This sounds like the method implemented to avoid using Linq, not sure why. So I am not sure we need this change, though I am OK if the area experts OK with this change

Reply:
That's what my comment said. `ImmutableArrayExtensions.FirstOrDefault` is in the `System.Linq` namespace but in the `System.Collections.Immutable` assembly. The `System.Linq` assembly is still not used.